corehq/apps/reports/views.py

```diff
@@ -2421,11 +2420,20 @@ def edit_form(request, domain, instance_id):
     instance = _get_location_safe_form(domain, request.couch_user, instance_id)
     assert instance.domain == domain
 
-    updates = {name: request.POST[name] for name in request.POST}
-    if FormProcessorInterface(domain).update_responses(instance, updates, request.couch_user.get_id):
-        messages.success(request, _('Question responses saved.'))
+    form_data, question_list_not_found = get_readable_data_for_submission(instance)
+    old_values, dummy = get_data_cleaning_data(form_data, instance)
+    updates = {name: request.POST[name] for name in request.POST
+            if name in old_values and old_values[name] != request.POST[name]}
```

Comment:
nit: could use iteritems for clarity and to avoid two hash lookups
```py
updates = {name: value for name, value in six.iteritems(request.POST)
           if name in old_values and old_values[name] != value}
```
I'm not sure if it makes sense in this context, but you may want to use [`request.POST.lists()`](https://stackoverflow.com/a/3303396/10840) here instead of `six.iteritems(request.POST)`.

Reply:
reworked this code while addressing another comment